lldb/source/Core/ValueObject.cpp

```diff
@@ -3015,11 +3017,12 @@ llvm::Expected<lldb::ValueObjectSP> ValueObject::CastDerivedToBaseType(
 
   lldb::TargetSP target = GetTargetSP();
   // The `value` can be a pointer, but GetChildAtIndex works for pointers too.
-  lldb::ValueObjectSP inner_value;
+  lldb::ValueObjectSP inner_value = GetSP();
 
   for (const uint32_t i : base_type_indices)
     // Force static value, otherwise we can end up with the "real" type.
-    inner_value = GetChildAtIndex(i, /*can_create_synthetic*/ false);
```

Comment:
I don't understand this change.  How is getting `this->GetSP()->GetChildAtIndex` different from `this->GetChildAtIndex`?

The other change here seems to be to change can_create_synthetic from false to true.  IIUC, the comment directly above this change says why it chose `false` here, though it is a somewhat confusing comment.  Since you seem to disagree with that comment, you should probably change the comment to say why you think `true` is the right value here.

Reply:
The difference is that now in each loop iteration the rhs is 'inner_loop->GetChildAtIndex...', which will be different each iteration through the loop; before it was just 'GetChildAtIndex', which did not change on each iteration.

re false->true: The original lldb-eval code on which that was based used 'false', but it was also making the call through the SB API interface (SBValues, SBTypes, etc). I found that the cast test cases passed with lldb-eval when this value was false, but failed in my translated, migrated version. Setting this to true allows those failing test cases to pass.

I don't have any easy way to add those test cases to this PR because they all use 'frame variable' (using my DIL implementation) for doing the casts...But I do have them, and I am running (and passing) them.

Reply:
Ah, I missed that I was looking at a diff against your original diff which didn't make sense.  In context the code seems fine.   

But then that also showed that you are the author of the "somewhat confusing comment".  Can you make that comment more helpful?  Suppressing the synthetic child production would force this child to get its type directly from the CompilerType, which I would call the "real" child.  So that comment seems to make sense when we passed `true` but doesn't make sense when passing true.

You don't need to introduce tests for this at this stage if that's dependent on other changes, but this seems like a tricky point, so it should have a comment.  But this one doesn't make sense to me.

Reply:
Done.